src/domain/message-formatter.ts

```diff
@@ -75,36 +87,81 @@ export default class MessageFormatter {
       return `${chalk.bold.blue(user)}: ${chalk.green('You sent a ' + getRavenMediaType(media))}`;
     }
     if (isRavenMessage(media)) {
-      const img = media.visual_media.media.image_versions2.candidates
-        .find(({ width }) => width === media.visual_media.media.original_width) || media.visual_media.media.image_versions2.candidates[0];
+      const img = media.visual_media.media.image_versions2?.candidates
+        .find(({ width }) => width === media.visual_media.media.original_width) || media.visual_media.media.image_versions2?.candidates[0];
+      const video = media.visual_media.media?.video_versions
+        ?.sort((a: Video, b: Video) => (a.height < b.height) ? 1 : -1)
+        ?.shift();
 
-      if (!img) {
-        return `${chalk.bold.blue(user)}: ${chalk.red('[Ephemeral picture]')}${chalk.bold.red('(no url)')}`;
+      if (!img && !video) {
+        return `${chalk.bold.blue(user)}: ${chalk.red('[Ephemeral media]')}${chalk.bold.red('(no url)')}`;
       }
 
-      const mediaAppearance = chalk.green(`[Ephemeral picture](${img.url})`)
-
+      let mediaAppearance = '';
+      if (img) {
+        mediaAppearance += chalk.green(`[Ephemeral image](${img.url})`);
+      }
+      if (video) {
+        mediaAppearance += chalk.magenta(`[Ephemeral video](${video.url})`);
+      }
       return `${chalk.bold.blue(user)}: ${mediaAppearance}`;
     }
 
     if (isRavenExpired(media)) {
       return `${chalk.bold.blue(user)}: ${chalk.green('Sent a ' + getRavenMediaType(media) + ' (expired)')}`;
     }
   
-    return `${chalk.bold.blue(user)}: ${chalk.red('[Ephemeral picture](unhandled type)')}`;
+    return `${chalk.bold.blue(user)}: ${chalk.red('[Ephemeral media](unhandled type)')}`;
   }
 
   private mediaFormatter(msg: DirectInboxFeedResponseItemsItem): string {
     const user = this.users[msg.user_id];
     const media = msg as MediaMessage;
 
-    const img = media.media.image_versions2.candidates
-      .find(({ width }) => width === media.media.original_width) || media.media.image_versions2.candidates[0];
+    const img = media.media.image_versions2?.candidates
+      .find(({ width }) => width === media.media.original_width) || media.media.image_versions2?.candidates[0];
+    const video = media.media?.video_versions
+      ?.sort((a: Video, b: Video) => (a.height < b.height) ? 1 : -1)
+      ?.shift();
```

Comment:
The optional chaining shouldn't be required here, it would already return earlier, and sort will return an array anyway
```suggestion
      .shift();
```